dts/riscv/rv32m1.dtsi

```diff
@@ -94,24 +94,188 @@
 			reg = <0x4101f000 0x88>;
 		};
 
-		intmux0: interrupt-controller@4004f000 {
+		intmux0: intmux@4004f000 {
 			compatible = "openisa,rv32m1-intmux";
-			#interrupt-cells = <1>;
-			interrupt-controller;
-			reg = <0x4004f000 0x20>;
+			reg = <0x4004f000 0x200>;
 			clocks = <&pcc0 0x13c>;
 			label = "INTMUX0";
 			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			ranges = <0x0 0x4004f000 0x200>;
+
+			intmux0_ch0: interrupt-controller@0 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH0_IRQ>;
+				reg = <0x0 0x40>;
+				label = "INTMUX0_CH0";
+				status = "disabled";
+			};
+
+			intmux0_ch1: interrupt-controller@40 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH1_IRQ>;
+				reg = <0x40 0x40>;
+				label = "INTMUX0_CH1";
+				status = "disabled";
+			};
+
+			intmux0_ch2: interrupt-controller@80 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH2_IRQ>;
+				reg = <0x80 0x40>;
+				label = "INTMUX0_CH2";
+				status = "disabled";
+			};
+
+			intmux0_ch3: interrupt-controller@c0 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH3_IRQ>;
+				reg = <0xc0 0x40>;
+				label = "INTMUX0_CH3";
+				status = "disabled";
+			};
+
+			intmux0_ch4: interrupt-controller@100 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH4_IRQ>;
+				reg = <0x100 0x40>;
+				label = "INTMUX0_CH4";
+				status = "disabled";
+			};
+
+			intmux0_ch5: interrupt-controller@140 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH5_IRQ>;
+				reg = <0x140 0x40>;
+				label = "INTMUX0_CH5";
+				status = "disabled";
+			};
+
+			intmux0_ch6: interrupt-controller@180 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH6_IRQ>;
+				reg = <0x180 0x40>;
+				label = "INTMUX0_CH6";
+				status = "disabled";
+			};
+
+			intmux0_ch7: interrupt-controller@1c0 {
+				compatible = "openisa,rv32m1-intmux-ch";
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				interrupts = <INTMUX_CH7_IRQ>;
+				reg = <0x1c0 0x40>;
+				label = "INTMUX0_CH7";
+				status = "disabled";
+			};
```

Comment:
Maybe this could be simplified with a macro that takes the index. Lots of typo-prone duplication.

Reply:
Maybe it'd need an address to.

Reply:
Noticed it gets messy with the C preprocessor because of the hash marks and stuff, so never mind.